src/nvim/ex_docmd.c

```diff
@@ -1201,6 +1201,19 @@ static void get_wincmd_addr_type(char_u *arg, exarg_T *eap)
   }
 }
 
+static char_u *skip_colon_white(const char_u *p, bool skipleadingwhite)
```

Comment:
I guess this could use a comment explaining what exactly it does.

Reply:
Sure, shall add :+1:

Reply:
Instead of handling more edge-cases, I think we need to use the full ex-cmd parsing.

Reply:
I've had a look, and we'd have to split out `do_one_cmd` - it parses the command line but has several side affects touching global state. I'll have a crack at pulling it out, containing the side effects and see where it takes us.

Reply:
Regarding this - I've had a go and I believe it's doable, but I'd prefer to merge this kind of change in a different PR (if at all). Shall update with the PR when it's up.